consensus/src/consensus_provider.rs

```diff
@@ -38,22 +40,29 @@ pub trait ConsensusProvider {
 pub fn make_consensus_provider(
     node_config: &mut NodeConfig,
     network_sender: ConsensusNetworkSender<Vec<SignedTransaction>>,
-    network_receiver: ConsensusNetworkEvents<Vec<SignedTransaction>>,
+    network_events: ConsensusNetworkEvents<Vec<SignedTransaction>>,
     executor: Arc<Mutex<Executor<LibraVM>>>,
     state_sync_client: Arc<StateSyncClient>,
     consensus_to_mempool_sender: mpsc::Sender<ConsensusRequest>,
     libra_db: Arc<dyn DbReader>,
+    reconfig_events: libra_channel::Receiver<(), OnChainConfigPayload>,
 ) -> Box<dyn ConsensusProvider> {
     let storage = Arc::new(StorageWriteProxy::new(node_config, libra_db));
     let txn_manager = Box::new(MempoolProxy::new(consensus_to_mempool_sender));
     let state_computer = Arc::new(ExecutionProxy::new(executor, state_sync_client));
-
-    Box::new(ChainedBftSMR::new(
+    let input = ChainedBftSMRInput {
```

Comment:
seems reasonable although ... why on earth do we even need this input? could we just merge this into ChainedBftSMR::new?

Reply:
yeah, one step a time but probably it's trivial to just convert over